src/reactviews/pages/QueryResult/table/plugins/autoColumnSize.plugin.ts

```diff
@@ -0,0 +1,423 @@
+            let change = false;
+            let headerElements: HTMLElement[] = [];
+            let columnDefs: Slick.Column<T>[] = [];
+            let colIndices: number[] = [];
+
+            for (let i = 0; i <= headerColumns.children.length; i++) {
+                let headerEl = jQuery(
+                    headerColumns.children.item(i)! as HTMLElement,
+                );
+                let columnDef = headerEl.data("column");
+                if (columnDef) {
+                    headerElements.push(headerEl[0]);
+                    columnDefs.push(columnDef);
+                    colIndices.push(this._grid.getColumnIndex(columnDef.id));
+                }
+            }
+
+            let headerWidths: number[] = this.getElementWidths(headerElements);
+            headerWidths = headerWidths.map((width) => {
+                return width + this._options.extraColumnHeaderWidth!;
+            });
+            let maxColumnTextWidths: number[] = this.getMaxColumnTextWidths(
+                columnDefs,
+                colIndices,
+            );
+
+            for (let i = 0; i < columnDefs.length; i++) {
+                let colIndex: number = colIndices[i];
+                let column: Slick.Column<T> = allColumns[colIndex];
+                let autoSizeWidth: number =
+                    Math.max(headerWidths[i], maxColumnTextWidths[i]) + 1;
+                if (autoSizeWidth !== column.width) {
+                    allColumns[colIndex].width = autoSizeWidth;
+                    change = true;
+                }
+            }
+
+            if (change) {
+                this.onPostEventHandler.unsubscribeAll();
+                this._grid.setColumns(allColumns);
+                this._grid.onColumnsResized.notify();
+            }
+        }
+    }
+
+    private handleDoubleClick(e: JQuery.TriggeredEvent<HTMLElement, unknown>) {
+        let headerEl = jQuery(e.currentTarget).closest(".slick-header-column");
+        let columnDef = headerEl.data("column");
+
+        if (!columnDef || !columnDef.resizable) {
+            return;
+        }
+
+        e.preventDefault();
+        e.stopPropagation();
+
+        this.resizeColumn(headerEl, columnDef);
+    }
+
+    private resizeColumn(headerEl: JQuery, columnDef: Slick.Column<T>) {
+        let headerWidth = this.getElementWidths([headerEl[0]])[0];
+        let colIndex = this._grid.getColumnIndex(columnDef.id!);
+        let origCols = this._grid.getColumns();
+        let allColumns = deepClone(origCols);
+        allColumns.forEach((col, index) => {
+            col.formatter = origCols[index].formatter;
+            col.asyncPostRender = origCols[index].asyncPostRender;
+        });
+        let column = allColumns[colIndex];
+
+        let autoSizeWidth =
+            Math.max(
+                headerWidth,
+                this.getMaxColumnTextWidth(columnDef, colIndex),
+            ) + 1;
+
+        // Only resize if the current width is smaller than the new width.
```

Comment:
In testing/playing around, this felt a bit funky (felt like a bug until I remembered that I'd seen a comment about it).  I know ADS does it this way, but Excel will resize both directions.

Reply:
Do you think I should change this to resize in both directions then?

Reply:
Yeah, let's allow it to resize in both directions.